### PR TITLE
Add delete repo and delete file method

### LINKED ISSUE #17: When resources are removed from a dataset, they should no longer appear in the `data` folder of that dataset on GitHub
When I remove a resource from a dataset, I want to see it was removed from `datapackage.json` as well as from the `data` folder so I know my dataset is being updated correctly.

## Acceptance

* [x] When I remove a resource from a dataset, I want to see it was removed from `datapackage.json` as well as from the `data` folder so I know my dataset is being updated as it should. **Done** see #18 

## Tasks

* [x] Add a delete file  method

## Analysis

* There is no resource here https://github.com/gift-data/costa-rica-presupuesto-ingresos-2020/blob/main/datapackage.json#L20 but they are still showing up at https://github.com/gift-data/costa-rica-presupuesto-ingresos-2020/tree/main/data

### PR DESCRIPTION
**Pull request by risenW**
This pull fixes #17

### CODE REVIEW

File: src/backend/github.js [hunk 1]
```diff
@@ -218,6 +225,51 @@ class GitHubStorage extends StorageBackend {
...
+   * @param {Object} options params from which repo is created, can be one of:
+   * options = {objectId: Unique repository name,
+   *            path: path of the file to delete, defaults to full repository,
+   *            message: commit message for files commited to the repo on github,
+   *            branch: main or master, branch on github to look for content
+   *            isResource: If you're deleting a metadata resource or not
+   */
+  async delete(options = { isResource: false }) {
+    return new Promise(async (resolve, reject) => {
+      let { objectId, path, branch, isResource } = options
```
Comment (cotts) [inline src/backend/github.js]: use `const` here

File: src/backend/github.js [hunk 2]
```diff
@@ -218,6 +225,51 @@ class GitHubStorage extends StorageBackend {
...
+      if (!objectId) {
+        throw new Error('objectId name cannot be null')
+      }
+      let { org, repoName } = this._parseId(objectId)
```
Comment (cotts) [inline src/backend/github.js]: use `const` here

**Review by cotts — CHANGES_REQUESTED** [review]

File: src/backend/github.js (hunk 2 above)
Comment (risenW) [reply src/backend/github.js]: Thanks

File: src/backend/github.js (hunk 1 above)
Comment (risenW) [reply src/backend/github.js]: Thanks

**Pull request merged**